OpenMage/magento-lts
# Do not calculate tax notifications every time

### Description

On a multi website/store/currencies, a notification can appear about tax:

![image](https://user-images.githubusercontent.com/31816829/200182050-814ee979-4ca8-4250-99f5-4995f62f9732.png)

Today, theses notifications are calculated on each page load.
I found with Blackfire this is about 120 ms (this depend of number of stores):

![notif](https://user-images.githubusercontent.com/31816829/200182123-6cbea187-b8b5-455e-af37-dfb1ddea7c8a.png)

This PR add 3 flags when Tax configuration is saved.
So the block is calculated only if necessary and only if not ignored.

![image](https://user-images.githubusercontent.com/31816829/200182291-51f39a07-7d0e-430a-8d58-63300fdda03f.png)

### Manual testing scenario

- Create 10+ store views (to see a difference).
- Go to System > Configuration > Tax > Calculation Settings.
- Set the following values (default config): _Total, Shipping Address, Including Tax, Including Tax, Including Tax, After discount, Including Tax, Custom price, Yes_.
- Save the configuration, there is no notice.
- Update _Apply Customer_ Tax to _Before Discount_.
- Save the configuration, there is a notice.
- Ignore/hide the notice.
- Save the configuration without changes, there is a notice.

With the PR or without the PR, the time to generate _Mage_Tax_Block_Adminhtml_Notifications_ does not change.
**But**, with the PR, when the block is ignored or not necessary, time to generate the block is negligible.

### Contribution checklist
